sql-assignment-answers

```diff
@@ -0,0 +1,1564 @@
+
+mysql> SELECT * FROM Customers WHERE Rating=100;
+
++------+---------+--------+--------+------+
+| CNUM | CNAME   | CITY   | RATING | SNUM |
++------+---------+--------+--------+------+
+| 2001 | Hoffman | London |    100 | 1001 |
+| 2006 | Clemens | London |    100 | 1001 |
+| 2007 | Pereira | Rome   |    100 | 1004 |
++------+---------+--------+--------+------+
+
+
+3) Find the largest order taken by each salesperson on each date.
+
+mysql> SELECT SNAME, ODATE, ONUM, AMT FROM Salespeople, Orders WHERE Salespeople.SNUM = Orders.SNUM AND AMT IN (SELECT MAX(AMT) FROM Orders GROUP BY ODATE);
+
++---------+------------+------+---------+
+| SNAME   | ODATE      | ONUM | AMT     |
++---------+------------+------+---------+
+| Serres  | 1990-03-10 | 3005 | 5160.45 |
+| Peel    | 1990-05-10 | 3008 | 4723.00 |
+| AxelRod | 1990-04-10 | 3009 | 1713.23 |
+| Peel    | 1990-06-10 | 3011 | 9891.88 |
++---------+------------+------+---------+
+
+
+4) Arrange the Order table by descending customer number. 
+
+mysql> SELECT * FROM Orders ORDER BY CNUM DESC;
+
++------+---------+------------+------+------+
+| ONUM | AMT     | ODATE      | CNUM | SNUM |
++------+---------+------------+------+------+
+| 3001 |   18.69 | 1990-03-10 | 2008 | 1007 |
+| 3006 | 1098.16 | 1990-03-10 | 2008 | 1007 |
+| 3002 |  1900.1 | 1990-03-10 | 2007 | 1004 |
+| 3008 | 4723.00 | 1990-05-10 | 2006 | 1001 |
+| 3011 | 9891.88 | 1990-06-10 | 2006 | 1001 |
+| 3007 |   75.75 | 1990-04-10 | 2004 | 1002 |
+| 3010 | 1309.95 | 1990-06-10 | 2004 | 1002 |
+| 3005 | 5160.45 | 1990-03-10 | 2003 | 1002 |
+| 3009 | 1713.23 | 1990-04-10 | 2002 | 1003 |
+| 3003 |  767.19 | 1990-03-10 | 2001 | 1001 |
++------+---------+------------+------+------+
+
+
+5) Find which salespeople currently have orders in the order table. 
+
+mysql> SELECT DISTINCT Orders.SNUM, Salespeople.SNAME FROM Orders INNER JOIN Salespeople ON Orders.SNUM = Salespeople.SNUM;
+
++------+---------+
+| SNUM | SNAME   |
++------+---------+
+| 1007 | Rifkin  |
+| 1004 | Motika  |
+| 1001 | Peel    |
+| 1002 | Serres  |
+| 1003 | AxelRod |
++------+---------+
+
+
+6) List names of all customers matched with the salespeople serving them. 
+
+mysql> SELECT c.CNAME, s.SNAME FROM Customers c, Salespeople s WHERE c.SNUM=s.SNUM;
+
++----------+---------+
+| CNAME    | SNAME   |
++----------+---------+
+| Hoffman  | Peel    |
+| Giovanni | AxelRod |
+| Liu      | Serres  |
+| Grass    | Serres  |
+| Clemens  | Peel    |
+| Pereira  | Motika  |
+| Cisneros | Rifkin  |
++----------+---------+
+
+
+
+7) Find the names and numbers of all salespeople who have more than one customer.
+
+
+mysql> SELECT SNUM,SNAME FROM Salespeople WHERE 1<(SELECT COUNT(*) FROM Customers WHERE Salespeople.SNUM = Customers.SNUM);
+
++------+--------+
+| SNUM | SNAME  |
++------+--------+
+| 1001 | Peel   |
+| 1002 | Serres |
++------+--------+
+
+
+
+8) Count the orders of each of the salespeople and output the results in descending order. 
```

Comment:
Edge case missed
Try to think about this 
if there is salespeople who is not associated with any order 
Expected
In that case the order count of salespeople  shows as 0.